From 3887ba07ccab4aa970c24a22a183b97e255e6ece Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Fri, 14 Apr 2023 14:37:23 +0200
Subject: [PATCH] Avoid C99 incompatibility around
 seahorse_pkcs11_backend_initialize

Move the prototype for seahorse_pkcs11_backend_initialize into the
header file, so C code generated from Vala sources will use it.
This avoids an implicit function declaration during the C-build
of src/application.vala, and resulting build failures with future
compilers.
---
Upstream PR: https://gitlab.gnome.org/GNOME/seahorse/-/merge_requests/214

 pkcs11/seahorse-pkcs11-backend.c | 2 --
 pkcs11/seahorse-pkcs11-backend.h | 2 ++
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/pkcs11/seahorse-pkcs11-backend.c b/pkcs11/seahorse-pkcs11-backend.c
index b7ea21e1f..31d7cb815 100644
--- a/pkcs11/seahorse-pkcs11-backend.c
+++ b/pkcs11/seahorse-pkcs11-backend.c
@@ -43,8 +43,6 @@ enum {
 	PROP_LOADED,
 };
 
-void  seahorse_pkcs11_backend_initialize (void);
-
 static SeahorsePkcs11Backend *pkcs11_backend = NULL;
 
 struct _SeahorsePkcs11Backend {
diff --git a/pkcs11/seahorse-pkcs11-backend.h b/pkcs11/seahorse-pkcs11-backend.h
index ddcfd2fcc..65efbd0a0 100644
--- a/pkcs11/seahorse-pkcs11-backend.h
+++ b/pkcs11/seahorse-pkcs11-backend.h
@@ -43,4 +43,6 @@ SeahorsePkcs11Backend *  seahorse_pkcs11_backend_get           (void);
 GcrCollection *          seahorse_pkcs11_backend_get_writable_tokens (SeahorsePkcs11Backend *self,
                                                                       gulong with_mechanism);
 
+void  seahorse_pkcs11_backend_initialize (void);
+
 #endif /* SEAHORSE_PKCS11_BACKEND_H_ */
-- 
GitLab

